postgres/osmap.yaml

```diff
@@ -91,9 +91,46 @@ RedHat:
 {% endif %}
 
 Suse:
+  pkg_repo:
+    name: pgdg-sles-{{ release }}
+    humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
+    #Using sles-12 upstream repo for opensuse
+    baseurl: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch'
+    key_url: 'https://download.postgresql.org/pub/repos/zypp/{{ repo.version }}/suse/sles-12-$basearch/repodata/repomd.xml.key'
+    gpgcheck: 1
+    gpgautoimport: True
+
+{% if repo.use_upstream_repo %}
+  {# Pillars needed are 'use_upstream_repo: True' and 'version: n.n'. #}
+  {# Avoid setting package names as pillars, as may corrupt postgres. #}
+  {% set lib_dir = '/var/lib/pgsql/' ~ repo.version ~ '/data' %}
+
+  pkg: postgresql{{ release }}-server
+  pkg_client: postgresql{{ release }}
+  conf_dir: {{ lib_dir }}
+   {% if release|int == 10 %}
+  service: postgresql-{{ release }}
+   {% endif %}
+  #This is postgresql-libs in defaults.yml but upstream is libpqxx
+  pkg_libpq_dev: libpqxx
+  pkg_dev: postgresql{{ release }}-devel
+  pkg_libs: postgresql{{ release }}-libs
+
+  prepare_cluster:
+    #Full path needed as initdb is NOT 'cross version compatible' binary
+    command: /usr/pgsql-{{ release }}/bin/initdb --pgdata='{{ lib_dir }}'
+    test: test -f '{{ lib_dir }}/PG_VERSION'
+
+  # Handles etc alternatives for 'cross version compatible' binaries
+  pkgs_extra: ['postgresql{{ release }}-contrib',]
```

Comment:
I've checked out the `postgresql10-contrib-10.1-1PGDG.sles12.x86_64.rpm` and appears it has nothing to do with alternatives at all.
I think we should disable extra packages installation by default, but may add a comment to `pillar.example` file mentioning how typically they called if having `use_upstream_repo` enabled.

Reply:
Thank you for checking. I fully agree extra package installation can be disabled by default for general OS mapping/defaults/pillar.example.

I will retest with Zypp upstream and if not needed will disabled the variable for Suse.